graveler/graveler.go

```diff
@@ -300,6 +311,10 @@ type VersionController interface {
 	// Compare returns the difference between the commit where 'to' was last synced into 'from', and the most recent commit of `from`.
 	// This is similar to a three-dot (from...to) diff in git.
 	Compare(ctx context.Context, repositoryID RepositoryID, from, to Ref) (DiffIterator, error)
+
+	// Hooks are used to register callbacks on specific actions (pre commit/merge)
+	// When a callback returns an error, the action will fail
+	Hooks() *Hooks
```

Comment:
I'd accept a context here for logging and cancellation as well as key this by a repository ID as hooks belong to a specific repo

Reply:
Graveler supplies the basic callback functionality we had in the cataloger.
There is no execution here, just getter.